### ISSUE TITLE
Cannot install, missing config

### ISSUE THREAD

**Issue by cammelspit** [ryanrudolfoba/SteamOS-Waydroid-Installer#42]
so, my issue is relatively straightforward. I cannot get the script to install because at a point after the packages have been installed the installer errors out by saying that it cannot find the configuration file. after Double checking, I have verified that the Clone repo did work properly and that all the files in the GitHub repo are present and accounted for. I could probably install this myself manually, which is what I intended on doing before I actually found this script. It seemed at the time like it would be a much easier way to go about installing Waydroid. I don't think it could be a configuration issue on my end however I do have brew and the nix package manager installed on my deck but that hasn't caused issues anywhere else. This is especially true considering that your script uses the AUR instead. If you have any ideas, let me know and I'll be more than happy to test for you. 

Thanks!

**Comment by ryanrudolfoba** [comment]
please provide screenshot of the error so i can see which part it broke. but most probably what happened is that the waydroid init failed (slow mirror etc etc). i put some sanity check in there maybe its not enough or gave a different return code.

**Comment by ryanrudolfoba** [comment]
the script does a sanity check - it checks if waydroid config is already present to determine if its a fresh install or not. if the config is not found, it gives the "Config file missing" and proceeds to download and initialze waydroid.

try this please. edit the script and add exit on line 254 -
![image](https://github.com/ryanrudolfoba/SteamOS-Waydroid-Installer/assets/98122529/b91b9d75-74cd-4d57-8815-f0b6220cb944)


run the script again so it exits immediately when it goes to the part "Config file missing"

then manually run -

sudo waydroid init -s GAPPS

what SteamOS version are you on?
and the error looks like a python error. i created and tested the script on stock SteamOS Stable / Preview.

**Comment by ryanrudolfoba** [comment]
the script does a sanity check - it checks if waydroid config is already present to determine if its a fresh install or not. if the config is not found, it gives the "Config file missing" and proceeds to download and initialze waydroid.

try this please. edit the script and add exit on line 254 -
![image](https://github.com/ryanrudolfoba/SteamOS-Waydroid-Installer/assets/98122529/4f02f6ba-ef69-4c5b-b184-e730b222a5fd)



run the script again so it exits immediately when it goes to the part "Config file missing"

then manually run -

sudo waydroid init -s GAPPS

what SteamOS version are you on?
and the error looks like a python error. i created and tested the script on stock SteamOS Stable / Preview.

**Comment by cammelspit** [comment]
Here is the what I got, I am also just running the current stable, here are the specifics. VERSION_ID=3.5.7 BUILD_ID=20231122.1

This is what happened when i tried to run that line manually.

`
steamos-waydroid-installer on  main [!] took 17s 
❯ sudo waydroid init -s GAPPS
Traceback (most recent call last):
  File "/usr/bin/waydroid", line 6, in <module>
    import tools
  File "/usr/lib/waydroid/tools/__init__.py", line 8, in <module>
    import dbus.mainloop.glib
ModuleNotFoundError: No module named 'dbus'
`

**Comment by ryanrudolfoba** [comment]
Looks like its a broken python on your system. Not sure how to fix. The script doesnt touch python it just uses what is already installed.

**Comment by cammelspit** [comment]
Well, thanks for that much at least. I at least have a place to start with here. If I recall, I installed a specific version of python to run a specific tool for flashing the firmware on one of my devices. Due to injury I have been almost entirely bed bound for nearly a year so I have been using my deck as my primary PC. It is likely I broke something. I need to do some research and figure out how I broke it and more importantly, how to restore it. Perhaps just swapping to the preview branch will work since it will reimage the base system. I will let you know if I figure it out and again, thanks!

**Comment by ryanrudolfoba** [comment]
quick work around -

Power off Steam Deck
While off, hold 3dots + power
Recovery menu will appear, select PREVIOUS
Once it boots up, perform an update to STABLE
It will oberwrite the Stable to the default packages

**Comment by cammelspit** [comment]
I got it working. It seems my install of master commander using Brew was to blame. It has python as a prerequisite and since brew works entirely in user land it naturally insyalls it's own python package. The issue here is that even though the one SteamOS has installed is still there and functional, i was running the script from an environment thay allows brew packages to work. Two options, uninstall mc, python, and glib from brew entirely or I can open a terminal that does not see the brew environemnt and run the script from there. Either way works. For simple troubleshooting I chose to uninstall the offending packages and your script worked perfectly. I will have to test and see how nix does things because it is almost Valve supported. Valve devs added a nix directory and the official nix installer is steam deck aware, it even persists after OS updates. I only just installed that a few days ago and if it doesn't break stuff like brew does I'll gladly switch.

Thanks a heap for your help here. It turns out it was one of the weird things I did to my own deck that broke it but your help troubleshooting it absolutely helped me get it fixed way faster than it would have taken me on my own.

![360_F_461512195_rBzpceWlGBFjOFr2zSK1Qe5obeY5pRCQ](https://github.com/ryanrudolfoba/SteamOS-Waydroid-Installer/assets/9344114/abafaec1-2d0e-4f30-bbc8-34e8bf599bbd)
